src/blant.c

```diff
@@ -1379,14 +1381,15 @@ int main(int argc, char *argv[])
 	    if(_confidence >= 1) _confidence /= 100; // user specified percent
 	    break;
 	case 'k': _k = atoi(optarg);
+		_k_base = MIN(_k, OLD_MAX_K);
 	    if (_GRAPH_GEN && _k >= 33) {
 		_k_small = _k % 10;
-		if (!(3 <= _k_small && _k_small <= 8))
-		    Fatal("%s\nERROR: k [%d] must be between 3 and 8\n%s", USAGE_SHORT, _k_small);
+		if (!(3 <= _k_small && _k_small <= MAX_K))
+		    Fatal("%s\nERROR: k [%d] must be between 3 and %d\n%s", USAGE_SHORT, _k_small, MAX_K);
 		_k /= 10;
 		assert(_k_small <= _k);
 	    } // First k indicates stamping size, second k indicates KS test size.
-	    if (!(3 <= _k && _k <= 8)) Fatal("%s\nERROR: k [%d] must be between 3 and 8\n%s", USAGE_SHORT, _k);
+	    if (!(3 <= _k && _k <= MAX_K)) Fatal("%s\nERROR: k [%d] must be between 3 and %d\n%s", USAGE_SHORT, _k, MAX_K);
```

Comment:
I need to understand what's happening here. It's very weird. Please add comments to describe why all this %10 and /10 weirdness is happening. You can describe it in Skype if you want, but if it stays in the code there MUST be comments describing what's happening.